app/src/main/java/com/infomaniak/mail/utils/PermissionUtils.kt

```diff
@@ -93,13 +123,31 @@ class PermissionUtils @Inject constructor(private val activity: FragmentActivity
      * Request storage permission only for Android API below 29.
      */
     fun requestDownloadManagerPermission(downloadCallback: () -> Unit) {
-        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.Q) storageForActivityResult?.launch(storagePermission)
         this.downloadCallback = downloadCallback
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.Q) storageForActivityResult?.launch(STORAGE_PERMISSION)
     }
     //endregion
 
     companion object {
+
+        @RequiresApi(Build.VERSION_CODES.TIRAMISU)
+        private const val POST_NOTIFICATIONS_PERMISSION = Manifest.permission.POST_NOTIFICATIONS
+        const val READ_CONTACTS_PERMISSION = Manifest.permission.READ_CONTACTS
         @get:DeprecatedSinceApi(Build.VERSION_CODES.Q, "Only used for DownloadManager below API 29")
-        private const val storagePermission = Manifest.permission.WRITE_EXTERNAL_STORAGE
+        private const val STORAGE_PERMISSION = Manifest.permission.WRITE_EXTERNAL_STORAGE
+
+        /**
+         * If the user has manually disabled notifications permissions, stop requesting it.
+         * Manually disabled means the permission was granted at one point, but is no more.
+         */
```

Comment:
Is this comment placed where it should?

Reply:
It should be here, but i'll also add it to new functions about notifications